plugins/nf-azure/src/main/nextflow/cloud/azure/file/AzBashLib.groovy

```diff
@@ -50,17 +51,17 @@ class AzBashLib extends BashFunLib<AzBashLib> {
         nxf_az_upload() {
             local name=\$1
             local target=\${2%/} ## remove ending slash
-            local base_name="\$(basename "$name")"
-            local dir_name="\$(dirname "$name")"
+            local base_name="\$(basename "\$name")"
+            local dir_name="\$(dirname "\$name")"
 
```

Comment:
This means is all broken what we have now?

Reply:
Not really, the change is basically to accommodate the interpolation for `--block-blob-tier $blobTier --block-size-mb $blockSize` (and other) options.

Reply:
ok, what confuse me is that I don't see in the PR the change from a `'''` string block to a `"""` string block

Reply:
Okay, we can avoid changing the string type and escaping the `$`.


> prepend a snippet that declares that values a Bash variables

Do you mean something like the following? 


```groovy



    protected String setupAzCopyOpts() {
        if( !retryMode )
            return ''
        """
        # custom env variables used for azcopy opts
        export AZCOPY_BLOCK_SIZE_MB=${blockSize}
        export AZCOPY_BLOCK_BLOB_TIER=${blobTier}
        """.stripIndent().rightTrim()
    }


    protected String azLib() {
        '''
            ...
                azcopy cp "$name" "$target?$AZ_SAS" --recursive --block-blob-tier $AZCOPY_BLOCK_BLOB_TIER --block-size-mb $AZCOPY_BLOCK_SIZE_MB
            ...

        '''.stripIndent()
    }


    String render() {
        super.render() + setupAzCopyOpts() + azLib()
    }




                
```

Reply:
yes!